src/Engine/ProtoCore/Executive.cs

```diff
@@ -15,9 +15,29 @@ public Executive (Core core)
 		}
 
         public ProtoCore.DSASM.Executive CurrentDSASMExec { get; set; }
-        public abstract bool Compile(out int blockId, ProtoCore.DSASM.CodeBlock parentBlock, ProtoCore.LanguageCodeBlock codeblock, ProtoCore.CompileTime.Context callContext, ProtoCore.DebugServices.EventSink sink = null, ProtoCore.AST.Node codeBlockNode = null, ProtoCore.AssociativeGraph.GraphNode graphNode = null);
-        public abstract StackValue Execute(int codeblock, int entry, ProtoCore.Runtime.Context callContext, ProtoCore.DebugServices.EventSink sink = null);
-        public abstract StackValue Execute(int codeblock, int entry, ProtoCore.Runtime.Context callContext, List<Instruction> breakpoints, ProtoCore.DebugServices.EventSink sink = null, bool fepRun = false);
+
+        public StackValue Execute(int codeblock, int entry, ProtoCore.Runtime.Context callContext, ProtoCore.DebugServices.EventSink sink)
+        {
+            if (!core.Options.CompileToLib)
+            {
+                ProtoCore.DSASM.Interpreter interpreter = new ProtoCore.DSASM.Interpreter(core);
+                CurrentDSASMExec = interpreter.runtime;
+                StackValue sv = interpreter.Run(codeblock, entry, CurrentDSASMExec.executingLanguage);
+                return sv;
+            }
+            else
+            {
+                return StackValue.Null;
+            }
+        }
+
+        public StackValue Execute(int codeblock, int entry, ProtoCore.Runtime.Context callContext, System.Collections.Generic.List<Instruction> breakpoints, ProtoCore.DebugServices.EventSink sink = null, bool fepRun = false)
+        {
+            ProtoCore.DSASM.Interpreter interpreter = new ProtoCore.DSASM.Interpreter(core, fepRun);
+            CurrentDSASMExec = interpreter.runtime;
+            StackValue sv = interpreter.Run(breakpoints, codeblock, entry, CurrentDSASMExec.executingLanguage);
+            return sv;
+        }
```

Comment:
similarly, what's the difference of these two ``Execute()``? Either comments or change to a better name, otherwise we have to look into the implementation to know what they do.